test/opexecuter.cpp

```diff
@@ -135,13 +135,26 @@ void OpExecuter::ExecuteOnTF() {
   DeactivateNGraph();
   ClientSession session(tf_scope_);
   ASSERT_EQ(Status::OK(), session.Run(sess_run_fetchoutputs_, &tf_outputs_));
+  for (int i = 0; i < tf_outputs_.size(); i++) {
+    NGRAPH_VLOG(5) << " TF op " << i << tf_outputs_[i].DebugString();
+  }
 }
 
 // Compares tf_outputs_ with ngraph_outputs_
 void OpExecuter::CompareNGraphAndTF() {
   ASSERT_EQ(tf_outputs_.size(), ngraph_outputs_.size());
   for (int i = 0; i < tf_outputs_.size(); i++) {
-    AssertTensorEquals(tf_outputs_[i], ngraph_outputs_[i]);
+    switch (expected_output_datatypes_[i]) {
+      case DT_FLOAT:
+        AssertTensorEquals<float>(tf_outputs_[i], ngraph_outputs_[i]);
+        break;
+      case DT_INT32:
+        AssertTensorEquals<int>(tf_outputs_[i], ngraph_outputs_[i]);
+        break;
+      default:
+        NGRAPH_VLOG(5) << "Could not find the corresponding function for the "
+                          "expected output datatype.";
+    }
```

Comment:
We should have an assert or fail (something like ```EXPECT_TRUE(false)```) it at this point, so that we take heed of the warning.
Otherwise given some other datatype CompareNGraphAndTF might happily exit without comparing, and we would not know (unless log-5 is enabled and someone is peering into it).

Reply:
Agree with @sayantan-nervana - need an ASSERT here to catch this condition.